feature/experimental/system/gnmi/benchmarking/ate_tests/internal/setup/setup.go

```diff
@@ -0,0 +1,200 @@
+	dutStartIPAddr = "192.0.2.1"
+	ateStartIPAddr = "192.0.2.2"
+	plenIPv4       = 30
+	authPassword   = "ISISAuthPassword"
+)
+
+var (
+	// DUTIPList is DUT IP list.
+	DUTIPList = make(map[string]net.IP)
+	// ATEIPList is ATE IP list.
+	ATEIPList = make(map[string]net.IP)
+)
+
+// BuildIPList builds list of ip addresses for the ports in binding file.
+// (Both DUT and ATE ports).
+func BuildIPList(dut *ondatra.DUTDevice) {
+	var dutIPIndex, ipSubnet, ateIPIndex int = 1, 2, 2
+	var endSubnetIndex = 253
+	for _, dp := range dut.Ports() {
+		dutNextIP := nextIP(net.ParseIP(dutStartIPAddr), dutIPIndex, ipSubnet)
+		ateNextIP := nextIP(net.ParseIP(ateStartIPAddr), ateIPIndex, ipSubnet)
+		DUTIPList[dp.ID()] = dutNextIP
+		ATEIPList[dp.ID()] = ateNextIP
+
+		// Increment DUT and ATE host ip index by 4.
+		dutIPIndex = dutIPIndex + 4
+		ateIPIndex = ateIPIndex + 4
+
+		// Reset DUT and ATE IP indexes when it is greater than endSubnetIndex.
+		if dutIPIndex > int(endSubnetIndex) {
+			ipSubnet = ipSubnet + 1
+			dutIPIndex = 1
+			ateIPIndex = 2
+		}
+	}
+
+}
+
+// nextIP returns ip address based on hostIndex and subnetIndex provided.
+func nextIP(ip net.IP, hostIndex int, subnetIndex int) net.IP {
+	s := ip.String()
+	sa := strings.Split(s, ".")
+	sa[2] = strconv.Itoa(subnetIndex)
+	sa[3] = strconv.Itoa(hostIndex)
+	s = strings.Join(sa, ".")
+	return net.ParseIP(s)
+}
+
+// BuildBenchmarkingConfig builds required configuration for DUT interfaces, ISIS and BGP.
+func BuildBenchmarkingConfig(t *testing.T) *oc.Root {
+	dut := ondatra.DUT(t, "dut")
+	d := &oc.Root{}
+
+	// Network instance and BGP configs.
+	netInstance := d.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
+
+	bgp := netInstance.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.As = ygot.Uint32(DUTAs)
+	global.RouterId = ygot.String(dutStartIPAddr)
+
+	afi := global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+	afi.Enabled = ygot.Bool(true)
+
+	pg := bgp.GetOrCreatePeerGroup(PeerGrpName)
+	pg.PeerAs = ygot.Uint32(ATEAs)
+	pg.PeerGroupName = ygot.String(PeerGrpName)
+
+	// ISIS configs.
+	isis := netInstance.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, ISISInstance).GetOrCreateIsis()
+
+	globalISIS := isis.GetOrCreateGlobal()
+	globalISIS.AuthenticationCheck = ygot.Bool(true)
+	globalISIS.Net = []string{fmt.Sprintf("%v.%v.00", DUTAreaAddress, DUTSysID)}
+	lspBit := globalISIS.GetOrCreateLspBit().GetOrCreateOverloadBit()
+	lspBit.SetBit = ygot.Bool(false)
+	isisTimers := globalISIS.GetOrCreateTimers()
+	isisTimers.LspLifetimeInterval = ygot.Uint16(600)
+	spfTimers := isisTimers.GetOrCreateSpf()
+	spfTimers.SpfHoldInterval = ygot.Uint64(5000)
+	spfTimers.SpfFirstInterval = ygot.Uint64(600)
+
+	isisLevel1 := isis.GetOrCreateLevel(1)
+	isisLevel1.Enabled = ygot.Bool(false)
+
+	isisLevel2 := isis.GetOrCreateLevel(2)
+	isisLevel2.Enabled = ygot.Bool(true)
```

Comment:
This leaf is soon to be deprecated in OpenConfig model, can we replace it with level-capability configuration?
ref. openconfig/public/pull/794

Reply:
@LimeHat : Thanks for the review. I have made the changes. Please take a look.